src/server/frontend_xwayland/xwayland_connector.cpp

```diff
@@ -37,6 +38,14 @@ void mf::XWaylandConnector::start()
 {
     if (wayland_connector->get_extension("x11-support"))
     {
+        /// TODO: make threadsafe
+        xwayland_spawner = std::make_unique<XWaylandSpawner>([this]()
+            {
+                if (xwayland_server && xwayland_spawner)
```

Comment:
1. Why add a TODO like this? Just do it.
2. Why are we setting `xwayland_spawner` and then testing it?